dali/base/dadfs.cpp

```diff
@@ -3512,6 +3526,15 @@ protected: friend class CDistributedFilePart;
         return fdesc.getClear();
     }
 
+    void setHistory(IFileDescriptor *fdesc)
+    {
+        IPropertyTree *history = fdesc->queryHistory();
+        if (isEmptyPTree(history))
+            resetHistory();
+        else
+            resetHistory(createPTreeFromIPT(history));
```

Comment:
Originally when everything was in one PR (core and dfuplus related) then was a createHistory() and eraseHistory() and you mentioned they are same functionality.
Now the two implementation is separated and I created this method which is used here to create  History  will be used in DFUPlus eraseHistory as well.